lib/common/strings.c

```diff
@@ -1171,6 +1171,33 @@ pcmk__strcmp(const char *s1, const char *s2, uint32_t flags)
     }
 }
 
+/*!
+ * \internal
+ * \brief Replace a dynamically allocated string
+ *
+ * If a given dynamically allocated string is different from another given
+ * string, free it and replace it with a duplicate of the new one. If the new
+ * one is NULL, set the original to NULL.
+ *
+ * \param[in] str    Pointer to dynamically allocated string
+ * \param[in] value  New value to duplicate
+ *
+ * \note The caller retains ownership of \p str.
+ */
+void
+pcmk__copy_or_null(char **str, const char *value)
```

Comment:
Yeah I wasn't happy with the name. Initially I thought of pcmk__str_replace() to parallel crm_xml_replace(). That's not explicit it's doing a copy, but there wouldn't be much use for a function that just did an assignment, so maybe that could work? Or pcmk__str_update()? Otherwise pcmk__strdup_or_null() would be fine ...